### PR TITLE
Set resource limits for connector definitions: persistence layer

### PR DESCRIPTION
**Pull request by cgardens**
relates to: https://github.com/airbytehq/airbyte/issues/10295
## What
* Adds columns in the db to store per connector definition resource requirements

## Recommended reading order
1. `x.java`
2. `y.python`

### CODE REVIEW

File: airbyte-config/models/src/main/resources/types/JobType.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+"$schema": http://json-schema.org/draft-07/schema#
+"$id": https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/JobType.yaml
+title: JobType
+description: enum that describes the different types of jobs that the platform runs.
+type: string
+# todo (cgardens) - sync naming with java
```
Comment (cgardens) [inline airbyte-config/models/src/main/resources/types/JobType.yaml:7]: reminder!

File: airbyte-config/models/src/main/resources/types/ActorDefinitionResourceRequirements.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+"$id": https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/ActorDefinitionResourceRequirements.yaml
+title: ActorDefinitionResourceRequirements
+description: actor definition specific resource requirements
+type: object
+additionalProperties: true
+properties:
+  default:
+    description: if set, these are the requirements that should be set for ALL jobs run for this actor definition.
+    "$ref": ResourceRequirements.yaml
+  jobSpecific:
```
Comment (cgardens) [inline airbyte-config/models/src/main/resources/types/ActorDefinitionResourceRequirements.yaml:12]: want to leave this open to handling requirement for other job types (e.g. check, or what now), but for right now the acute need is for sync jobs.

File: airbyte-bootloader/src/test/java/io/airbyte/bootloader/BootloaderAppTest.java [hunk 3]
```diff
@@ -80,7 +80,7 @@ void testBootloaderAppBlankDb() throws Exception {
         mockedConfigs.getConfigDatabaseUrl())
             .getAndInitialize();
     val configsMigrator = new ConfigsDatabaseMigrator(configDatabase, this.getClass().getName());
-    assertEquals("0.35.28.001", configsMigrator.getLatestMigration().getVersion().getVersion());
+    assertEquals("0.35.32.001", configsMigrator.getLatestMigration().getVersion().getVersion());
```
Comment (cgardens) [inline airbyte-bootloader/src/test/java/io/airbyte/bootloader/BootloaderAppTest.java:83]: @davinchia why do i need to change this test every time there is a migration? seems very manual.

File: airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_32_001__AddConnectorDefinitionResourceLimits.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  @Override
+  public void migrate(final Context context) throws Exception {
+    LOGGER.info("Running migration: {}", this.getClass().getSimpleName());
+
+    final DSLContext ctx = DSL.using(context.getConnection());
+    addTombstoneColumn(ctx);
+  }
+
+  public static void addTombstoneColumn(final DSLContext ctx) {
```
Comment (cgardens) [inline airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_35_32_001__AddConnectorDefinitionResourceLimits.java:27]: todo fix method name.

File: airbyte-config/models/src/main/resources/types/JobType.yaml (hunk 1 above)
Comment (cgardens) [reply airbyte-config/models/src/main/resources/types/JobType.yaml:7]: will switch it to use the same names and casing as this:
```
public enum TemporalJobType {
  GET_SPEC,
  CHECK_CONNECTION,
  DISCOVER_SCHEMA,
  SYNC,
  RESET_CONNECTION,
  CONNECTION_UPDATER,
  REPLICATE
}
```

File: airbyte-bootloader/src/test/java/io/airbyte/bootloader/BootloaderAppTest.java (hunk 3 above)
Comment (davinchia) [reply airbyte-bootloader/src/test/java/io/airbyte/bootloader/BootloaderAppTest.java:83]: automatically updating expected tests always felt somewhat wrong to me. I don't feel too strongly so I'm happy to add this if you think it's alright.

**Review by lmossman — APPROVED** [review]
Once the TODOs are taken care of, this all makes sense to me

**Pull request merged**